### PR TITLE
WT-10333 Fix timestamp simulator coverity issues.

### PR DESCRIPTION
**Pull request by Siddhartha8899**

### CODE REVIEW

File: test/simulator/timestamp/src/session_simulator.cpp [hunk 1]
```diff
@@ -68,9 +68,12 @@ session_simulator::begin_transaction(const std::string &config)
     /* Check if the read or prepared timestamp should be rounded up. */
     auto pos = config_map.find("roundup_timestamps");
     if (pos != config_map.end()) {
-        if (pos->second.find("read=true"))
+        size_t found = pos->second.find("read=true");
+        if (found != std::string::npos)
```
Comment (ajmorton) [inline test/simulator/timestamp/src/session_simulator.cpp:72]: nit: If we're not using found anywhere else we can just do `if (pos->second.find("read=true") != std::string::npos)`

File: test/simulator/timestamp/call_log_manager/call_log_manager.cpp [hunk 2]
```diff
@@ -231,7 +231,8 @@ call_log_manager::call_log_query_timestamp(const json &call_log_entry)
         session_simulator *session = get_session(session_id);
 
         ret = session->query_timestamp(config, hex_ts);
-    }
+    } else
```
Comment (ajmorton) [inline test/simulator/timestamp/call_log_manager/call_log_manager.cpp:234]: nit: Our other examples of exception raising look like `throw std::invalid_argument("error_message")`

File: test/simulator/timestamp/src/session_simulator.cpp (hunk 1 above)
Comment (Siddhartha8899) [reply test/simulator/timestamp/src/session_simulator.cpp:72]: Makes sense, thanks!

File: test/simulator/timestamp/call_log_manager/call_log_manager.cpp (hunk 2 above)
Comment (Siddhartha8899) [reply test/simulator/timestamp/call_log_manager/call_log_manager.cpp:234]: This looks like a better approach, thanks!

File: test/simulator/timestamp/src/session_simulator.cpp (hunk 1 above)
Comment (Siddhartha8899) [reply test/simulator/timestamp/src/session_simulator.cpp]: Updated in [ad38239](https://github.com/wiredtiger/wiredtiger/pull/8599/commits/ad38239fe7e3bea3396aec6934b5d43b6c2d2c2e)

File: test/simulator/timestamp/call_log_manager/call_log_manager.cpp (hunk 2 above)
Comment (Siddhartha8899) [reply test/simulator/timestamp/call_log_manager/call_log_manager.cpp:234]: Updated in [b684e52](https://github.com/wiredtiger/wiredtiger/pull/8599/commits/b684e527bda0724e4c15f7b713e232f50babf4ef)

**Review by ajmorton — APPROVED** [review]
LGTM. Thanks!

**Review by Sean04 — APPROVED** [review]
lgtm

**Pull request merged**